### ISSUE TITLE
Remove unnecessary `if` statement when results can be determined at compile time

### ISSUE THREAD

**Issue by poopsicles** [rust-lang/rust-clippy#11378]
### What it does

If an `if else` expression only contains static types, such as `i8`s and `u32`s, not variables, then it's likely that one branch can be totally removed without the "correctness" of the code changing.

A Reddit thread ([Archive.org link](https://web.archive.org/web/20230821181400/https://www.reddit.com/r/rust/comments/15x91cs/just_realized_ifs_can_be_nested_in_a_different_way/?rdt=44732)) demonstrated this by putting an `if else` inside another `if`, which was evaluated correctly.

I'm not exactly sure whether this lint should capture:

- Just freestanding `if` statements that can be determined ahead (comparing two integers)
- `if else` statements (which most definitely yield a value) inside another `if` statement (like the Reddit example)

### Advantage

- Remove unnecessary code branches.
- Makes the code easier to read.

### Drawbacks

Someone might intend to later "fill out" the other branches, which is why this is limiting it to `if-else` expressions in another `if` statement, as opposed to freestanding `if` statements might make more sense, the former is usually an unintended, while the latter could be intended to be added to.

### Example

```rust
let a = 13;
if if a == 13 { 10 } else { 0 } > 5 { // <- nested if-else
    println!("This will be printed");
}
```

Could be written as this (if we're just looking for nested ones):

```rust
let a = 13;
if 10 > 5 { // <- inner if-else removed, however the outer if-statement, which *could* be removed, isn't
    println!("This will be printed");
}
```

Or

```rust
let a = 13;
println!("This will be printed"); // <- if removed, but most likely more "annoying"
```

**Comment by poopsicles** [comment]
Another example would be

```rust
if 1 > 0 {
    println!("hi");
}
```

This example just deals with static ints, and not any variables since that might require the compiler probably?

**Comment by poopsicles** [comment]
Perhaps simply lifting the inner if-else is more achieveable?

Going from:

```rust
let a = 13;
if if a == 13 { 10 } else { 0 } > 5 { // <- nested if-else
    println!("This will be printed");
}
```

To:

```rust
let a = 13;
let thing = if a == 13 { 10 } else { 0 };

if thing > 5 {
    println!("This will be printed");
}
```
